### ISSUE TITLE
events-based API

### ISSUE THREAD

**Issue by hjdivad** [heimdalljs/heimdalljs-lib#31]
Here's a possible API that lets us use an events-based format

``` js
let nodeToken = heimdall.start('constant:string:colon:delimited');

// set on the current node
// we do this rather than have a label POJO to avoid the allocation cost when we don't have anything more involved than a 
heimdall.setArgs({
  moduleName: 'whatever arbitrary thing you want'
});

// tokens back so we can use them as indexes into a large array or for now just allocate an array per node
let [x, y, z ] = heimdall.registerMonitor('monitor-namespace', 'x', 'y', 'z');

// we need to benchmark this: we're adding function calls to reduce allocation costs.
heimdall.incStat(x);
heimdall.setStat(y, 200);



// starting a node with its own schema
//    we would need to benchmark whether or not we actually need to have a separate function for this rather than handling multiple signatures in `start`
let [token, x, y, z] = heimdall.startWithStats('foo:bar', 'x', 'y', 'z');
heimdall.incStat(x);
// ...
heimdall.stop(token);
```

Some things that we need to think about before committing to this:
- [ ] #19 make sure this doesn't affect async.  I'm pretty sure none of this does
- [ ] #13 we want to still support the problem that #13 aims to address.  Perhaps `registerMonitor` accepts a callback that is run to add the additional stats

**Comment by hjdivad** [comment]
cc @stefanpenner @runspired

**Comment by stefanpenner** [comment]
in broad strokes looks good

**Comment by runspired** [comment]
This landed as 0.3
